uxarray/grid.py

```diff
@@ -228,6 +259,118 @@ def __init_ds_var_names__(self):
+
+            # Present Coordinate Names
+            if self.ds.coords is not None:
+                if value in self.ds.coords:
+                    setattr(self, key, self.ds[value])
+
+    def integrate(self, var_key, quadrature_rule="triangular", order=4):
+        """ Integrates over all the faces of the given mesh.
+        Parameters
+        ----------
+
+        var_key : string, required
+            Name of variable for integration.
+
+        Returns
+        -------
+
+        double: integration result.
+
+        Examples
+        --------
+
+        Open grid file only
+        >>> grid = ux.open_dataset("grid.ug", "centroid_pressure_data_ug")
+
+
+        Open grid file along with data
+        >>> integral_psi = grid.integrate("psi")
+        """
+        integral = 0.0
+
+        # call function to get area of all the faces as a np array
+        face_areas = self.get_face_areas(quadrature_rule, order)
+
+        face_vals = self.ds.get(var_key).to_numpy()
+        integral = np.dot(face_areas, face_vals)
+
+        return integral
+
+    def get_face_areas(self, quadrature_rule="triangular", order=4):
+        """Face area calculation property for grid class, calculates area of
+        all faces in the mesh.
+
+        Returns
+        -------
+
+        area of all the faces in the mesh. : ndarray
+
+        Examples
+        --------
+
+        Open a uxarray grid file
+
+        >>> grid = ux.open_dataset("/home/jain/uxarray/test/meshfiles/outCSne30.ug")
+
+        Get area of all faces in the same order as listed in grid.ds.Mesh2_face_nodes
+
+        >>> grid.get_face_areas
+        array([0.00211174, 0.00211221, 0.00210723, ..., 0.00210723, 0.00211221,
+            0.00211174])
+        """
+        if self._face_areas is None:
+            # area of a face call needs the units for coordinate conversion if spherical grid is used
+            coords_type = "spherical"
+            if not "degree" in self.ds.Mesh2_node_x.units:
+                coords_type = "cartesian"
+
+            face_nodes = self.ds.Mesh2_face_nodes.data
+            dim = self.ds.Mesh2.attrs['topology_dimension']
+
+            # initialize z
+            z = np.zeros((self.ds.nMesh2_node.size))
+
+            # call func to cal face area of all nodes
+            x = self.ds[self.ds_var_names["Mesh2_node_x"]].data
+            y = self.ds[self.ds_var_names["Mesh2_node_y"]].data
+            # check if z dimension
+            if self.ds.Mesh2.topology_dimension > 2:
+                z = self.ds[self.ds_var_names["Mesh2_node_z"]].data
+
+            # call function to get area of all the faces as a np array
+            self._face_areas = get_all_face_area_from_coords(
+                x, y, z, face_nodes, dim, quadrature_rule, order, coords_type)
+
+        return self._face_areas
+
+    # use the property keyword for declaration on face_areas property
+    face_areas = property(get_face_areas)
```

Comment:
AH didn't realize you want this to be configurable.

Would a user ever want to rewrite the face areas on the _same_ Grid object if they changed their mind about the quadrature and order? Presently this is not possible because the cache check is in `get_face_areas`

If you did want that then I would think about
1. Renaming `get_face_areas` to `compute_face_areas`
2. And following the caching pattern in `face_areas` as recommended earlier
``` python
@property
def face_areas(self): 
    if self._face_areas is None:
	self.compute_face_areas()
    return self._face_areas
```
3. Remove the caching check in `compute_face_areas`

Then a user can overwrite the cache by explicitly calling `grid.compute_face_areas` with their favourite options. And then rest of the code which uses `grid.face_areas` will use the updated one.

Reply:
^ Also this seems a somewhat minor interface issue. I agree with merging and resolving later.